driver/xrt/include/accl.hpp

```diff
@@ -235,35 +453,126 @@ class ACCL {
                        bool run_async = false,
                        std::vector<CCLO *> waitfor = {});
 
+  /**
+   * Check if ACCL is being run in simulated mode or not.
+   *
+   * @return true  ACCL is running on an emulator or simulator.
+   * @return false ACCL is running on hardware.
+   */
   bool is_simulated() const { return sim_mode; }
 
+  /**
+   * Construct a new buffer object without an existing host buffer.
+   *
+   * @tparam dtype              Datatype of the buffer.
+   * @param length              Amount of elements to allocate for.
+   * @param type                ACCL datatype of the buffer.
+   * @return std::unique_ptr<Buffer<dtype>> The allocated buffer.
+   */
   template <typename dtype>
-  std::unique_ptr<Buffer<dtype>> create_buffer(dtype *host_buffer,
-                                               size_t length, dataType type,
+  std::unique_ptr<Buffer<dtype>> create_buffer(size_t length, dataType type) {
+    return create_buffer<dtype>(length, type, _devicemem);
+  }
+
+  /**
+   * Construct a new buffer object without an existing host buffer on the
+   * specified memory bank.
+   *
+   * Only use this function if you want to store the buffer on a different
+   * memory bank than the devicemem bank specified during construction.
+   *
+   * @tparam dtype              Datatype of the buffer.
+   * @param length              Amount of elements to allocate for.
+   * @param type                ACCL datatype of the buffer.
+   * @param mem_grp             Memory bank to allocate buffer on.
+   * @return std::unique_ptr<Buffer<dtype>> The allocated buffer.
+   */
+  template <typename dtype>
+  std::unique_ptr<Buffer<dtype>> create_buffer(size_t length, dataType type,
                                                unsigned mem_grp) {
     if (sim_mode) {
-      return std::unique_ptr<Buffer<dtype>>(
-          new SimBuffer<dtype>(host_buffer, length, type,
-                               static_cast<SimDevice *>(cclo)->get_context()));
+      return std::unique_ptr<Buffer<dtype>>(new SimBuffer<dtype>(
+          length, type, static_cast<SimDevice *>(cclo)->get_context()));
     } else {
       return std::unique_ptr<Buffer<dtype>>(new FPGABuffer<dtype>(
-          host_buffer, length, type, device, (xrt::memory_group)mem_grp));
+          length, type, device, (xrt::memory_group)mem_grp));
     }
-    return std::unique_ptr<Buffer<dtype>>(nullptr);
   }
 
+  /**
+   * Construct a new buffer object from an existing host pointer.
```

Comment:
Maybe also mention here, that this function currently doesn't work with Vitis sw_emu? I know you mentioned the basic problem in SimBuffer::bo(), but here it may be way more likely to be spotted by a confused user.

Reply:
Agreed, will make it more clear.